pkg/frontend/querymiddleware/remote_read.go

```diff
@@ -282,6 +282,16 @@ func (r *remoteReadQueryRequest) GetQuery() string {
 	return r.promQuery
 }
 
+func (r *remoteReadQueryRequest) GetParsedQuery() parser.Expr {
+	if r.promQuery != "" {
+		expr, err := parser.ParseExpr(r.promQuery)
+		if err == nil {
```

Comment:
I think this is not expected to error because we generate the PromQL.
Perhaps someone can think of a better way to check?